app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/MapDataWithEditsSource.kt

```diff
@@ -78,14 +78,18 @@ class MapDataWithEditsSource internal constructor(
             val modifiedDeleted = ArrayList<ElementKey>()
             synchronized(this) {
                 /* We don't want to callOnUpdated if none of the changes affects map data provided
-                 * by MapDataWithEditsSource.
+                 * by MapDataWithEditsSource
                  * This is the case if
                  *  * All keys in deleted are already in deletedElements.
                  *  * The modified versions of all elements in updated are the same before and after
                  *    rebuildLocalChanges, except for the timestamp (expected to have few ms
                  *    difference) and version (never updated locally).
+                 * However, we do want to callOnUpdated if an updated element does not have an associated
+                 * modification, because in that case we cannot know whether the updated version is changed
+                 * compared to the previous version.
```

Comment:
```suggestion
                 *   * No new elements are being added
```

Well, this is `MapDataWithEditsSource`. It is not unexpected that there is map data without a modification. E.g. new elements could be added. (Only because we know the internals on how the whole system works together we can say that this usually doesn't happen except for this special case, but for `MapDataWithEditsSource` it is certainly not relevant how the whole system works, but only relevant what's on the interface)